addons/stock/models/product.py

```diff
@@ -126,6 +126,16 @@ def _compute_quantities(self):
         res = products._compute_quantities_dict(self._context.get('lot_id'), self._context.get('owner_id'), self._context.get('package_id'), self._context.get('from_date'), self._context.get('to_date'))
         for product in products:
             product.update(res[product.id])
+        for product in self:
+            no_availability_locations = product.mapped('stock_quant_ids.location_id').filtered(lambda l: l.no_availability)
```

Comment:
@temo-odoo
do we need this `no_availability_locations` here? as any way in the next line you have added the "no_availability" condition.